import_dts.py

```diff
@@ -462,11 +462,6 @@ def load(operator, context, filepath,
 
     # Then put objects in the armatures
     for obj in shape.objects:
-        if obj.node == -1:
-            print('Warning: Object {} is not attached to a node, ignoring'
-                  .format(shape.names[obj.name]))
-            continue
-
```

Comment:
I'm not sure if we should be letting all objects through. Below this, we're using `obj.node` to index into some arrays and if it ever equals -1 it will cause unexpected behavior. It looks like that this if statement was added in https://github.com/bansheerubber/io_scene_dts/commit/69183be0df497d3f3a8a9efbfb0c98fa0145d495, and after looking at the commit I'm thinking that we should re-introduce the if statement that was removed on line 444. I'm not entirely familiar with what `add_vertex_groups` does, but I'm assuming that due to the addition `if obj.node == -1` that a precondition of entering `add_vertex_groups` is `obj.node ==-1`, so lets re-introduce the if statement and also have it enclose the `add_vertex_groups` call so the resulting code looks like this:
```python
...

if ob.node != -1:
  add_vertex_groups(mesh, bobj, shape)

  if use_armature:
    bobj.parent = root_ob
    bobj.parent_bone = bone_names[obj.node]
    bobj.parent_type = "BONE"
    bobj.matrix_world = shape.nodes[obj.node].mat

    if mtype == Mesh.SkinType:
      modifier = bobj.modifiers.new('Armature', 'ARMATURE')
      modifier.object = root_ob
  else:
    bobj.parent = node_obs[obj.node]

...
```

Reply:
Here's some models that demonstrates the issue. You'll notice that the plugin currently imports a bunch of nodes but no objects. I'm not an expert on how the Torque game engine works in general, but I think the reason these files are like this is because the model is meant to be deformed by an armature rather than each object being parented to just one node. From what I can tell `add_vertex_groups` creates vertex groups in Blender for the bone weights for the armature. It does have it's own check for `if node != -1` so there shouldn't be any issues there with passing in objects without a node. 

[models.zip](https://github.com/bansheerubber/io_scene_dts/files/9184739/models.zip)